tests/storage/datavolume.go

```diff
@@ -347,6 +352,31 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 				}
 				storageClass, err = virtClient.StorageV1().StorageClasses().Create(context.Background(), storageClass, metav1.CreateOptions{})
 				Expect(err).ToNot(HaveOccurred())
+
+				dv = libstorage.NewRandomDataVolumeWithRegistryImportInStorageClass(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine), util.NamespaceTestDefault, storageClass.Name, k8sv1.ReadWriteOnce, k8sv1.PersistentVolumeFilesystem)
+				vmiDisk := v1.Disk{
+					Name: diskName,
+					DiskDevice: v1.DiskDevice{
+						Disk: &v1.DiskTarget{
+							Bus: v1.DiskBusVirtio,
+						},
+					}}
+				vmiVolume = v1.Volume{
+					Name: diskName,
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+								ClaimName: dv.ObjectMeta.Name,
+							},
+						},
+					}}
+				vmi = libvmi.New(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithResourceMemory("32Mi"),
+					libvmi.WithDisk(vmiDisk),
```

Comment:
As a would-be user of `WithDisk`, I find it a bit cumbersome to use. Can it be a bit more opinionated, having two arguments, `name` and `bus`? Or just `name`, and use the build pattern to allow a future `WithBus()` function modify the target bug?

Reply:
Hey @dankenigsberg ,
Thanks for reviewing.
I totally agree but in my opinion we should create another library for this ( `libdisk`) 
and use it within `libvmi` something like:
```
libvmi.new(
With..(...) ,
With..(...) ,
...
WithDisk(
      libdisk.new(...)
)
...
With..(...) ,
)
```
What do you think?

Reply:
I don't really understand this new API. Why wouldn't I want to use a more convenient option introduced here https://www.officeholidays.com/holidays/czech-republic/jan-hus-day

Reply:
@dankenigsberg , @xpivarc 
Thanks for the notes i didn't know about these functions
Modifed.